consensus/core/src/block_manager.rs

```diff
@@ -359,6 +419,66 @@ impl BlockManager {
         None
     }
 
+    /// Tries to unsuspend any blocks for the latest gc round. If gc round hasn't changed then no blocks will be unsuspended due to
+    /// this action.
+    pub(crate) fn try_unsuspend_blocks_for_latest_gc_round(&mut self) {
+        let _s = monitored_scope("BlockManager::try_unsuspend_blocks_for_latest_gc_round");
+        let gc_enabled = self.dag_state.read().gc_enabled();
+        let gc_round = self.dag_state.read().gc_round();
+        let mut total_blocks_gced = 0;
+
+        if !gc_enabled {
+            trace!("GC is disabled, no blocks will attempt to get unsuspended.");
+            return;
+        }
+
+        while let Some((block_ref, _children_refs)) = self.missing_ancestors.first_key_value() {
+            // If the first block in the missing ancestors is higher than the gc_round, then we can't unsuspend it yet. So we just put it back
+            // and we terminate the iteration as any next entry will be of equal or higher round anyways.
+            if block_ref.round > gc_round {
+                return;
+            }
+
+            total_blocks_gced += 1;
+
+            assert!(!self.suspended_blocks.contains_key(block_ref), "Block should not be suspended, as we are causally GC'ing and no suspended block should exist for a missing ancestor.");
+
+            // Also remove it from the missing list - we don't want to keep looking for it.
+            self.missing_blocks.remove(block_ref);
+
+            // Find all the children blocks that have a dependency on this one and try to unsuspend them
+            let unsuspended_blocks = self.try_unsuspend_children_blocks(*block_ref);
+
+            unsuspended_blocks.iter().for_each(|block| {
+                if block.round() <= gc_round {
```

Comment:
Should this be filtering unsuspended blocks that have been gc'd?

Reply:
hhm could you please elaborate a bit on this?

Reply:
How I am understanding this is that we have found a block in `missing_ancestors` that is garbage collected so we then `try_unsuspend_children_blocks` with this block no longer "missing". That will potentially return a set of blocks that have been unsuspended. You are then checking if any of those unsuspended blocks are <= gc_round which means they too should be garbage collected. 

So my confusion is why you only update the `total_blocks_gced` count but then accept these unsuspended blocks with `verify_block_timestamps_and_accept`?

Reply:
Oh got it. Ok maybe then the naming doesn't convey well the intention, but practically here I would like to capture "how many blocks under the gc_round have been accepted". Yes we do accept them in the end - which we could easily not to - but they can still be considered gc'ced. I can probably rename the variable to something like `blocks_accepted_below_gc_round`